src/wagtail_ai/ai/echo.py

```diff
@@ -0,0 +1,47 @@
+import random
+import time
+from collections.abc import Generator, Iterator
+from typing import Any, NotRequired, TypedDict
+
+from .base import AIBackend, AIResponse
+
+
+class EchoResponse(AIResponse):
+    def __init__(self, response_iterator: Iterator[str]) -> None:
+        self.response_iterator = response_iterator
+
+    def __iter__(self) -> Iterator[str]:
+        return self.response_iterator
+
+    def text(self) -> str:
+        return " ".join(self.response_iterator)
```

Comment:
**Thought:** We could make sure the iterator content gets saved somewhere because I guess it'll get exhausted... We don't want weird side-effects even if it's a test backend.